pulp_puppet_plugins/pulp_puppet/plugins/importers/metadata.py

```diff
@@ -169,20 +156,18 @@ def _extract_non_standard_json(filename, temp_dir):
         tgz = tarfile.open(name=filename)
         tgz.extractall(path=extraction_dir)
         tgz.close()
-    except Exception, e:
+    except Exception:
         raise InvalidTarball(filename), None, sys.exc_info()[2]
 
     try:
-        # Recursively look for the metadata file
-        metadata_file_dir = _find_file_in_dir(extraction_dir, constants.MODULE_METADATA_FILENAME)
-
-        if metadata_file_dir is None:
+        # Attempt to find the metadata in the first folder of the tar.gz
+        module_dir = os.listdir(extraction_dir)[0]
```

Comment:
From the docs on ``os.listdir``: "The list is in arbitrary order." So I think we need a different way to determine the root directory of the module.